.github/workflows/vulnerability-scan.yml

```diff
@@ -0,0 +1,48 @@
+name: Security vulnerability scan
+
+on:
```

Comment:
I hope to add PR level checking.
if a PR has new dependency added, we can know if the new dependency has vulnerability or not.

Reply:
It could be checked on every PR but I have not found this to add much value. Usually when a new dependency is added it is at **latest** version and has no vulnerability identified. Typically vulnerabilities appear in existing dependency versions so a nightly scan is effective at identifying that early. Having vulnerabilities in existing dependencies flagged up in all PR builds just gives a false impression that PRs are not good and potentially holds up delivering other critical fixes.